lib/chef/mixin/notifying_block.rb

```diff
@@ -0,0 +1,54 @@
+#--
+# Author:: Lamont Granquist <lamont@chef.io>
+# Copyright:: Copyright 2010-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class Chef
+  module Mixin
+    module NotifyingBlock
+
+      def notifying_block(&block)
+        begin
+          subcontext = subcontext_block(&block)
+          Chef::Runner.new(subcontext).converge
+        ensure
+          new_resource.updated_by_last_action(
```

Comment:
So a thing I've never fixed in Poise but we probably should here is that using `notifying_block` twice where the first block updates and the second doesn't ends up with `updated_by_last_action` being false. We should probably make this more like `updated_by_last_action(true) if <the condition>` so we don't overwrite an existing true with a false.